src/patternfly/components/Icon/icon.scss

```
@@ -0,0 +1,149 @@
+    --pf-c-icon--Height: var(--pf-c-icon--m-inline--Height);
+    --pf-c-icon__content--FontSize: var(--pf-c-icon--m-inline__content--FontSize);
+    --pf-c-icon__content--Color: currentcolor;
+
+    line-height: 1;
+
+    .pf-c-spinner {
+      --pf-c-spinner--diameter: 1em;
+    }
+  }
+
+  // Size modifiers
+  &.pf-m-sm {
+    --pf-c-icon--Width: var(--pf-c-icon--m-sm--Width);
+    --pf-c-icon--Height: var(--pf-c-icon--m-sm--Height);
+    --pf-c-icon__content--FontSize: var(--pf-c-icon--m-sm--content--FontSize);
+  }
+
+  &.pf-m-md {
+    --pf-c-icon--Width: var(--pf-c-icon--m-md--Width);
+    --pf-c-icon--Height: var(--pf-c-icon--m-md--Height);
+    --pf-c-icon__content--FontSize: var(--pf-c-icon--m-md--content--FontSize);
+  }
+
+  &.pf-m-lg {
+    --pf-c-icon--Width: var(--pf-c-icon--m-lg--Width);
+    --pf-c-icon--Height: var(--pf-c-icon--m-lg--Height);
+    --pf-c-icon__content--FontSize: var(--pf-c-icon--m-lg--content--FontSize);
+  }
+
+  &.pf-m-xl {
+    --pf-c-icon--Width: var(--pf-c-icon--m-xl--Width);
+    --pf-c-icon--Height: var(--pf-c-icon--m-xl--Height);
+    --pf-c-icon__content--FontSize: var(--pf-c-icon--m-xl--content--FontSize);
+  }
+
+  &.pf-m-in-progress {
+    --pf-c-icon__child--Opacity: 0;
+    --pf-c-icon__progress--Opacity: 1;
+  }
+}
+
+// Content and progress both have size modifiers
+.pf-c-icon__content,
+.pf-c-icon__progress {
+  font-size: var(--pf-c-icon__content--FontSize);
+
+  svg {
+    vertical-align: var(--pf-c-icon__content--svg--VerticalAlign);
+  }
+
+  // Size modifiers
+  &.pf-m-sm {
+    --pf-c-icon__content--FontSize: var(--pf-c-icon--m-sm--content--FontSize);
+  }
+
+  &.pf-m-md {
+    --pf-c-icon__content--FontSize: var(--pf-c-icon--m-md--content--FontSize);
+  }
+
+  &.pf-m-lg {
+    --pf-c-icon__content--FontSize: var(--pf-c-icon--m-lg--content--FontSize);
+  }
+
+  &.pf-m-xl {
+    --pf-c-icon__content--FontSize: var(--pf-c-icon--m-xl--content--FontSize);
+  }
+}
+
+// Content element
+.pf-c-icon__content {
+  color: var(--pf-c-icon__content--Color);
+  opacity: var(--pf-c-icon__child--Opacity, 1);
+
+  // Status modifiers
+  &.pf-m-danger {
+    --pf-c-icon__content--Color: var(--pf-c-icon__content--m-danger--Color);
+  }
+
+  &.pf-m-warning {
+    --pf-c-icon__content--Color: var(--pf-c-icon__content--m-warning--Color);
+  }
+
+  &.pf-m-success {
+    --pf-c-icon__content--Color: var(--pf-c-icon__content--m-success--Color);
+  }
+
+  &.pf-m-info {
+    --pf-c-icon__content--Color: var(--pf-c-icon__content--m-info--Color);
+  }
+
+  &.pf-m-default {
+    --pf-c-icon__content--Color: var(--pf-c-icon__content--m-default--Color);
+  }
+}
+
+// Progress element
+.pf-c-icon__progress {
+  position: absolute;
+  top: calc(50% + calc(.5 * var(--pf-c-icon__content--svg--VerticalAlign)));
+  opacity: var(--pf-c-icon__progress--Opacity, 0);
+  transform: translateX(0) translateY(calc(-50% - calc(.5 * var(--pf-c-icon__content--svg--VerticalAlign))));
```

do you need to define `translateX()`?

```suggestion
  transform: translateY(calc(-50% - calc(.5 * var(--pf-c-icon__content--svg--VerticalAlign))));
```